data/Drug_target_interaction/transform.py

```diff
@@ -0,0 +1,132 @@
+import pandas as pd
+import yaml
+from tdc.multi_pred import DTI
+
+
+def get_and_transform_data():
+    # get raw data
+    data = DTI(name="BindingDB_Kd")
+    fn_data_original = "data_original.csv"
+    data.get_data().to_csv(fn_data_original, index=False)
+
+    # create dataframe
+    df = pd.read_csv(
+        fn_data_original,
+        delimiter=",",
+    )  # not necessary but ensure we can load the saved data
+
+    # check if fields are the same
+    fields_orig = df.columns.tolist()
+    assert fields_orig == [
+        "Drug_ID",
+        "Drug",
+        "Target_ID",
+        "Target",
+        "Y",
+    ]
+
+    # overwrite column names = fields
+    fields_clean = [
+        "compound_name",
+        "SMILES",
+        "target_name",
+        "Target_aa",
+        "binding",
+    ]
+    df.columns = fields_clean
+
+    # data cleaning
+    """
+    df.compound_name = (
+        df.compound_name.str.strip()
+    )  # remove leading and trailing white space characters
+    """
+    assert not df.duplicated().sum()
+
+    # save to csv
+    fn_data_csv = "data_clean.csv"
+    df.to_csv(fn_data_csv, index=False)
+
+    # create meta yaml
+    meta = {
+        "name": "Drug-Target Interaction",  # unique identifier, we will also use this for directory names
+        "description": """The activity of a small-molecule drug is measured by its binding affinity with the target protein.
+        Given a new target protein, the very first step is to screen a set of potential compounds to find their activity.
+        Traditional method to gauge the affinities are through high-throughput screening wet-lab experiments.
+        However, they are very expensive and are thus restricted by their abilities to search over a large set of candidates
+        Drug-target interaction prediction task aims to predict the interaction activity score in silico given only the accessible compound structural information and protein amino acid sequence.""",
+        "targets": [
+            {
+                "id": "binding",  # name of the column in a tabular dataset
+                "description": "small-molecule protein interaction.",  # description of what this column means
+                "units": "Kd",  # units of the values in this column (leave empty if unitless)
+                "type": "continous",  # can be "categorical", "ordinal", "continuous"
+                "names": [  # names for the property (to sample from for building the prompts)
```

Comment:
```suggestion
		"uris" : ["	http://purl.obolibrary.org/obo/NCIT_C20604"]
                "names": [  # names for the property (to sample from for building the prompts)
```

Do you think the URI https://bioportal.bioontology.org/ontologies/BAO?p=classes&conceptid=http://purl.obolibrary.org/obo/NCIT_C20604 makes sense?

Reply:
This still needs to be added!